projects/kit/components/slider/slider-key-steps.directive.ts

```diff
@@ -96,7 +98,18 @@ export class TuiSliderKeyStepsDirective
             return;
         }
 
-        this.slider.value = this.transformToNativeValue(controlValue);
+        const clampedControlValue = clamp(controlValue, this.min, this.max);
+
+        if (controlValue !== clampedControlValue) {
+            this.control?.setValue(clampedControlValue);
```

Comment:
Components should not affect control value unless through user actions. Form control component is a GUI for user to work with control, we can correct use actions but not externally set value. If you want — you can keep the assertion but manipulating form control value like that is something we avoid as it is philosophically wrong.

Reply:
fixed ✅